fpsd/classify.py

```diff
@@ -0,0 +1,277 @@
+        self.frac_obs = world["observed_fraction"]
+        self.n_cores = n_cores
+        self.k = k
+        self.feature_scaling = feature_scaling
+        self.db = database.ModelStorage()
+
+    def get_dict(self):
+        """Get config options as a dict"""
+
+        return {"hyperparameters": self.hyperparameters,
+                "model_type": self.model_type,
+                "world_type": self.world_type,
+                "observed_world_fraction": self.frac_obs,
+                "numfolds": self.k,
+                "train_class_balance": 'DEFAULT',
+                "base_rate": 'DEFAULT'}
+
+    def train_single_fold(self, train_x, train_y):
+        """Trains a model and saves it as self.trained_model
+
+        Args:
+            train_x [ndarray]: features in training set (no id column, no target)
+            train_y [ndarray]: target variable in training set (no id column)
+        """
+
+        print("Training {} classifier with {}".format(self.model_type,
+                                                      self.hyperparameters))
+
+        modelobj = self._get_model_object(self.model_type,
+                                          self.hyperparameters,
+                                          self.n_cores)
+        trained_model = modelobj.fit(train_x, train_y)
+        return trained_model
+
+    def score(self, test_x, trained_model):
+        """Generates continuous risk scores for a testing set.
+
+        Args:
+            test_x [ndarray]: testing features
+            trained_model [sklearn object]: trained classifier object
+
+        Returns:
+            result_y [ndarray]: predictions on test set
+        """
+
+        result_y = trained_model.predict_proba(test_x)
+        score_positive_class = result_y[:, 1]
+        return score_positive_class
+
+    def train_eval_all_folds(self, x_val, y_val):
+        """Trains and evaluates models over all folds.
+
+        Args:
+            timestamp [string]: timestamp of the time the model is run,
+                                used as model identifier
+            x_val [ndarray]: feature matrix
+            y_val [ndarray]: label vector
+
+        Returns:
+            auc [float]: area under the ROC curve
+            mean_fpr [list of floats]: false positive rate averaged over
+                                       all kfolds
+            mean_tpr [list of floats]: true positive rate averaged over
+                                       all kfolds
+            trained_model [sklearn object]: trained object to be pickled
+                                            so that it can be used for
+                                            scoring 
+        """
+
+        if self.world_type == "closed":
+            # Why we use stratified k-fold here:
+            # http://stats.stackexchange.com/questions/49540/understanding-stratified-cross-validation
+            cv = cross_validation.StratifiedKFold(y_val, n_folds=self.k,
```

Comment:
Not sure how this would address #61, but it’s not the best cross validation method to use here as it we want to make sure we are preserving the class balance as well as the fact that we’d have to specify the size of the training set, which `cross_validation.StratifiedKFold` does automatically by our specifying `k`